### DISCUSSION TITLE
Strange hyper error with store.put

### DISCUSSION THREAD

**Discussion by CorinJG** [apache/arrow-rs#5350] [Q&A] [answered]
When I use the API like this, I get the error below:

```rust
                    let store = object_store::aws::AmazonS3Builder::from_env()
                        .with_bucket_name(bucket)
                        .with_region(region)
                        .build()?;

                    store.put(&target_path, bytes).await?;
```

Error:

> [2024-01-30T23:17:25Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:29Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:32Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:35Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:38Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:41Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:45Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:48Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:51Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:54Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> [2024-01-30T23:17:58Z DEBUG reqwest::connect] starting new connection: http://169.254.169.254/
> Generic { store: "S3", source: Error { retries: 10, message: "request error", source: Some(reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Ipv4(169.254.169.254)), port: None, path: "/latest/api/token", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 113, kind: HostUnreachable, message: "No route to host" })) }), status: None } }

It tries a bunch of times to connect to this particular IP address and then fails producing an error code about host unreachable. I googled and understand this IP is some range associated with aws meta data? It looks like a DNS error but no idea why my machine can't get DHCP to work only with using this crate?

**Comment by tustvold** [comment, marked as answer]
It is trying to talk to an EC2 metadata endpoint to source credentials, this won't work on a local machine. If you provide it with credentials it should behave

https://docs.rs/object_store/latest/object_store/aws/struct.AmazonS3Builder.html#example

Alternatively you can disable signing if credentials are not required - https://docs.rs/object_store/latest/object_store/aws/struct.AmazonS3Builder.html#method.with_skip_signature
